rezz.lic

```diff
@@ -1,118 +1,131 @@
-      when 'You sense a spirit nearby, but you are unable to make it out clearly.'
-        next
-      when /You clearly sense the spirit of #{person} in the Void/i
-        break
-      when 'Your sense of sight has not fully transitioned to the spirit world'
-        pause 2
-        next
-      when "You don't have enough harnessed mana to infuse that much."
-        next
-      else
-        next
-      end
+  
+  def rejuv(quick = false)
+    quick ? begin @rejuv['cambrinth'] = []; @rejuv['prep_time'] = 5; @rejuv['mana'] = 5 end : nil
+    Flags.add('wrong-name', "You can't cast that at yourself")
+    Flags.add('silver', 'A thin silver nimbus surrounds')
+    Flags.add('alive', 'Nothing happens')
+    until Flags['silver'] || Flags['alive'] || Flags['wrong-name']
+	  pause 1 while DRStats.mana < 50
+      cast_spell(@rejuv, @settings)
+  	  pause 0.5
     end
+    Flags['alive'] ? begin echo "*** #{@person} is already alive, dumbass! ***"; exit end : nil
+    Flags['wrong-name'] ? begin echo "*** Name spelled incorrectly. ***"; exit end : nil
+    Flags.reset('rejuv-silver')
   end
-
-  def is_silver?(person)
-    /As you concentrate, a thin (.*) nimbus/ =~ bput("perc #{person}", 'As you concentrate, a thin (.*) nimbus')
-    color = Regexp.last_match(1)
-    echo "Memories are #{color}"
-    waitrt?
-    color.eql? 'silver'
-  end
-
-  def rejuv(person)
-    return if is_silver?(person)
-    data = get_data('spells').spell_data['Rejuvenation']
-    data['cast'] = "cast #{person}"
-    if @rejuv_mana.zero?
-      data = check_discern(data, @settings)
-      @rejuv_mana = data['mana']
+    
+  def cast_rezz
+    Flags.add('rezz-ready', 'You prepare to look for the lost spirits of the dead')
+    Flags.add('rezz-not-prepped', 'This pattern may only be cast with full preparation')
+    cast_spell(@rezz, @settings)
+    if Flags['rezz-not-prepped']
+      while Flags['rezz-not-prepped'] 
+	    Flags.reset('rezz-not-prepped')
+        pause 1
+	    bput('cast', get_data('spells').cast_messages, )
+	  end
     end
-
-    loop do
-      cast_spell(data, @settings)
-      break if is_silver?(person)
+    while !Flags['rezz-ready']
+      pause 0.5
     end
   end
-
-  def check_buffs
-    wait_for_script_to_complete('buff', ['rezz'])
-  end
-
-  def skip_harness?
-    attunement = DRSkill.getrank('Attunement')
-    return false if attunement < 541
-    return true if attunement > 600
-    return true if check_if_buff_active('Persistence of Mana', 6)
-    false
-  end
-
-  def cast_rezz
-    data = get_data('spells').spell_data['Resurrection']
-    data = check_discern(data, @settings)
-    cast_spell(data, @settings)
-    data['mana']
+  
+  def find_soul
+    Flags.add('soul-found', /sense the spirit of #{@person} in the Void/i)
+    while !Flags['soul-found']
+      pause 1 while DRStats.mana < 50
```

Comment:
Previously this value was 40, and it still was not enough for my 50th cleric. I think 40 makes sense, but can argue for an even lower threshold based on my previous experience.

Reply:
Id argue 10 is fine. You want to keep harness/infusing until you find the soul or its no longer feasible to maintain rezz. Waiting to restore mana while you have a cyclic going thats draining it faster than you restore it will just make the user kill the script.

Reply:
Whatever everyone thinks is best.